File: app/src/main/java/org/deiverbum/app/repository/TodayRepository.java
package org.deiverbum.app.repository;

import org.deiverbum.app.data.source.remote.network.ApiService;

import javax.inject.Inject;

/**
 * <p>Repositorio de datos para el módulo Homilías.</p>
 * <p>Orden de búsqueda: </p>
 * <ul>
 *     <li>Firebase</li>
 *     <li>Api</li>
 * </ul>
 * @author A. Cedano
 * @version 1.0
 * @since 2022.1
 */

public class TodayRepository {
    final ApiService apiService;

    @Inject
    public TodayRepository(ApiService apiService) {
        this.apiService = apiService;
    }

}

File: app/src/main/java/org/deiverbum/app/domain/repository/TodayRepository.kt
package org.deiverbum.app.domain.repository

import org.deiverbum.app.domain.model.TodayRequest
import org.deiverbum.app.domain.model.TodayResponse

/**
 * <p>Interfaz para el repositorio del módulo Today.</p>
 *
 * @author A. Cedano
 * @since 2023.1.3
 */
interface TodayRepository {

    suspend fun getToday(todayRequest: TodayRequest): TodayResponse

}